klient/wait/conditions/conditions.go

```diff
@@ -58,6 +60,120 @@ func (c *Condition) ResourceScaled(obj k8s.Object, scaleFetcher func(object k8s.
 	}
 }
 
+// ResourceMatch is a helper function used to check if the resource under question has met a pre-defined state. This can
+// be leveraged for checking fields on a resource that may not be immediately present upon creation.
+func (c *Condition) ResourceMatch(obj k8s.Object, matchFetcher func(object k8s.Object) bool) apimachinerywait.ConditionFunc {
+	return func() (done bool, err error) {
+		if err := c.resources.Get(context.TODO(), obj.GetName(), obj.GetNamespace(), obj); err != nil {
```

Comment:
😢  My apologies for not catching this in earlier reviews.
The `Condition` methods, that act on the resources, should receive context (similar to how methods on `Resources` work):

```go
func (c *Condition) ResourceMatch(ctx context.Context, obj k8s.Object, matchFetcher func(object k8s.Object) bool) apimachinerywait.ConditionFunc {...}
```

That way, when a resource method is invoked, the resource can be passed in (instead of creating a `context.TODO()`):

```go
if err := c.resources.Get(ctx, obj.GetName(), obj.GetNamespace(), obj); err != nil {...}
```

Reply:
I am leaning towards returning a ConditionFunc that simply exits with the appropriate error

Reply:
Looking at this a little further, before I diverge the function signature from the rest of the package.

It seems like we could pass in `WithStopChannel(ctx.Done())` to `wait.For` and solve this, instead of having to add to the condition function here? 

```go
wait.For(conditions.New(resources).ResourceDeleted(...), WithStopChannel(ctx.Done()))
```

Reply:
@crandles I would rather pass in the context when the condition method is invoked instead of capturing the context too early. That way, it allows users of the method to adjust the context at any point (where as if it's captured early, then it cannot be modified).

Suggestion:
For sake of getting this PR in (i'd like a release soon with all the goodness you contributed), let's make that signature correction in a different PR. 

What do you think ?

Reply:
That makes sense to me -- we can adjust all of the signatures to include context at that point.